sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java

```diff
@@ -622,13 +627,118 @@ private void editSample(SampleDto sample, LabMessageDto labMessage) {
 		}
 
 		Window window = VaadinUiUtil.createPopupWindow();
+
+		// button configuration
+		addReferOrLinkToOtherLabButton(sampleEditComponent, sampleController.getDiseaseOf(sample), labMessage, window);
+		addReferredFromButton(sampleEditComponent, labMessage, window);
+
 		sampleEditComponent.addCommitListener(window::close);
 		sampleEditComponent.addDiscardListener(window::close);
 
 		showFormWithLabMessage(labMessage, sampleEditComponent, window, I18nProperties.getString(Strings.headingEditSample), false);
 
 	}
 
+	private void addReferOrLinkToOtherLabButton(
```

Comment:
seems to be a duplicate of `SampleController::addReferOrLinkToOtherLabButton`

Reply:
This method is different from the `SampleController::addReferOrLinkToOtherLabButton`. It includes handling a lab message.

In other words, it replicates the behaviour of `SampleController::addReferOrLinkToOtherLabButton` while processing lab messages.

One could probably write a method in the SampleController that works for both, normal sample editing and lab message processing. I decided against this, because it would add solely lab message related dependencies to the SampleController. Also the complexity of the SampleController would be increased for a quite specific purpose.

Knowing this, is the current approach acceptable for you?

Reply:
Adding Runnable or Consumer argument(s) on `SampleController::addReferOrLinkToOtherLabButton` to have different handlers would not increase the complexity but instead there won't be duplicated code for the same logic.

Reply:
I am not (yet) convinced. The problem I see is that Runnables would depend on an if statement in the method.
Consumers additionally would need a different amount of parameters for sample editing and lab message processing.
Maybe we should wait for a third opinion from @vidi42 ?

Reply:
I'm not sure. It looks like `SampleController::addReferOrLinkToOtherLabButton` could be made a bit more generic with using Runnables or Consumers so that it can be called from the `LabMessageController`, however the whole logic seems to be in the `if` statement and `ClickListeners` so I'm not sure if this will be simplified in the end.

@FredrikSchaeferVitagroup maybe you can try to make it more generic and if it looks ok to you then you can leave it like that, otherwise I wouldn't insist on making it generic.